sgv2-dynamo-api/src/main/java/io/stargate/sgv2/dynamosvc/dynamo/QueryProxy.java

```diff
@@ -0,0 +1,243 @@
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class QueryProxy extends Proxy {
+
+  private static final Logger logger = LoggerFactory.getLogger(QueryProxy.class);
+
+  private static final Map<String, Predicate> operatorConverter =
+      new HashMap<String, Predicate>() {
+        {
+          put("=", Predicate.EQ);
+          put("<=", Predicate.LTE);
+          put("<", Predicate.LT);
+          put(">=", Predicate.GTE);
+          put(">", Predicate.GT);
+        }
+      };
+
+  /**
+   * Basic partition key equality pattern
+   *
+   * <p>Example:<br>
+   * partitionKeyName = :partitionkeyval
+   */
+  private static final Pattern EQUALITY_PATTERN =
+      Pattern.compile("^\\s*(\\S+)\\s*=\\s*(:\\S+)\\s*$");
+
+  /**
+   * Basic partition key equality pattern + Sort key operation comparison pattern
+   *
+   * <p>Examples:<br>
+   * partitionKeyName = :partitionkeyval AND sortKeyName = :sortkeyval<br>
+   * partitionKeyName = :partitionkeyval AND sortKeyName < :sortkeyval<br>
+   * partitionKeyName = :partitionkeyval AND sortKeyName <= :sortkeyval<br>
+   * partitionKeyName = :partitionkeyval AND sortKeyName > :sortkeyval<br>
+   * partitionKeyName = :partitionkeyval AND sortKeyName >= :sortkeyval<br>
+   */
+  private static final Pattern OP_COMPARISON_PATTERN =
+      Pattern.compile("(?i)(\\S+)\\s*=\\s*(:\\S+)\\s*AND\\s*(\\S+)\\s*([<>=]+)\\s*(:\\S+)");
+
+  /**
+   * Basic partition key equality pattern + Sort key interval comparison pattern
+   *
+   * <p>Example:<br>
+   * partitionKeyName = :partitionkeyval AND sortKeyName BETWEEN :sortkeyval1 AND :sortkeyval2
+   */
+  private static final Pattern BETWEEN_PATTERN =
+      Pattern.compile(
+          "(?i)(\\S+)\\s*=\\s*(:\\S+)\\s*AND\\s*(\\S+)\\s*BETWEEN\\s*(:\\S+)\\s*AND\\s*(:\\S+)");
+
+  /**
+   * Basic partition key equality pattern + Sort key begins_with pattern (cannot use with number)
+   *
+   * <p>Example:<br>
+   * partitionKeyName = :partitionkeyval AND begins_with ( sortKeyName, :sortkeyval )
+   */
+  private static final Pattern BEGIN_WITH_PATTERN =
+      Pattern.compile(
+          "(?i)(\\S+)\\s*=\\s*(:\\S+)\\s*AND\\s*begins_with\\s*\\(\\s*(\\S+),\\s*(:\\S+)\\s*\\)");
+
+  public QueryResult query(QueryRequest queryRequest, StargateBridgeClient bridge) {
+    // Get table schema
+    final String tableName = queryRequest.getTableName();
+    Schema.CqlTable table =
+        bridge
+            .getTable(KEYSPACE_NAME, tableName)
+            .orElseThrow(() -> new IllegalArgumentException("Table not found"));
+    QueryOuterClass.ColumnSpec partitionKey = table.getPartitionKeyColumns(0);
+    List<QueryOuterClass.ColumnSpec> clusteringKeys = table.getClusteringKeyColumnsList();
+    Optional<QueryOuterClass.ColumnSpec> clusteringKey = Optional.empty();
+    if (CollectionUtils.isNotEmpty(clusteringKeys)) {
+      assert clusteringKeys.size() == 1;
```

Comment:
If this needs actual validation better check and throw regular exception? (if it should never occur, assert fine)

Reply:
It never occurs, unless our code is buggy (When creating the table, we ensure there is only one clustering key). This is more of a hint for developers.